controller/api/destination/watcher/k8s.go

```diff
@@ -50,26 +50,39 @@ func getHostAndPort(authority string) (string, Port, error) {
 // GetServiceAndPort is a utility function that destructures an authority into
 // a service and port.  If the authority does not represent a Kubernetes
 // service, an error is returned.  If no port is specified in the authority,
-// the HTTP default (80) is returned as the port number.
-func GetServiceAndPort(authority string) (ServiceID, Port, error) {
+// the HTTP default (80) is returned as the port number.  If the authority
+// is a pod DNS name then the pod hostname is also returned as the 3rd return
+// value.  See https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/.
+func GetServiceAndPort(authority string) (ServiceID, Port, string, error) {
 	host, port, err := getHostAndPort(authority)
 	if err != nil {
-		return ServiceID{}, 0, err
+		return ServiceID{}, 0, "", err
 	}
 	domains := strings.Split(host, ".")
-	// S.N.svc.cluster.local
-	if len(domains) != 5 {
-		return ServiceID{}, 0, fmt.Errorf("Invalid k8s service %s", host)
-	}
 	suffix := []string{"svc", "cluster", "local"}
-	for i, subdomain := range domains[2:] {
+	n := len(domains)
+	if n < 5 {
```

Comment:
should this also check for `n > 6` ? that way we could ditch the the last return statement.

Reply:
Yes, but I think we'd need to make the `n == 6` case into the `else` block.  I think the current way is clearer.